agenta-cli/agenta/sdk/decorators/llm_entrypoint.py

```diff
@@ -198,51 +267,67 @@ async def execute_function(self, func: Callable[..., Any], *args, **func_params)
             For synchronous functions, it calls them directly, while for asynchronous functions,
             it awaits their execution.
             """
+            data = None
+            trace = None
+
+            token = None
+            if tracing_context.get() is None:
+                token = tracing_context.set(TracingContext())
+
             is_coroutine_function = inspect.iscoroutinefunction(func)
-            start_time = time.perf_counter()
 
             if is_coroutine_function:
                 result = await func(*args, **func_params["params"])
             else:
                 result = func(*args, **func_params["params"])
 
-            end_time = time.perf_counter()
-            latency = end_time - start_time
+            if token is not None:
+                # check that it doesn't affect the tracing.tree
```

Comment:
Have you confirmed this, or still a to-do? If you haven't, how do I confirm it doesn't affect the tracing.tree?

Reply:
On it. Solved with `copy.deepcopy`, tested with a `print(tracing_context.get().tree)` after calling `dump_trace()`.

Reply:
Awesome, thank you.